### ISSUE TITLE
请问mlatransformer应该如何使用呢

### ISSUE THREAD

**Issue by huilin16** [PaddlePaddle/PaddleSeg#1441]
我使用contrib中的mlatransformer配置文件会报错，并且找不到SwinTransformer预训练模型路径

**Comment by michaelowenliu** [comment]
@huilin16 Hi, could you please show your config file?

**Comment by huilin16** [comment]
batch_size:&nbsp;1
iters:&nbsp;160000

model:
&nbsp;&nbsp;type:&nbsp;MLATransformer
&nbsp;&nbsp;pretrained:&nbsp;null&nbsp;#pretrain/pretrained_swin.pdparams
&nbsp;&nbsp;backbone:
&nbsp;&nbsp;&nbsp;&nbsp;type:&nbsp;SwinTransformer_base_patch4_window7_224
&nbsp;&nbsp;&nbsp;&nbsp;ape:&nbsp;False
&nbsp;&nbsp;&nbsp;&nbsp;drop_path_rate:&nbsp;0.3
&nbsp;&nbsp;&nbsp;&nbsp;patch_norm:&nbsp;True
&nbsp;&nbsp;num_classes:&nbsp;9
&nbsp;&nbsp;in_channels:&nbsp;[128,&nbsp;256,&nbsp;512,&nbsp;1024]
&nbsp;&nbsp;mlahead_channels:&nbsp;128

train_dataset:
&nbsp;&nbsp;type:&nbsp;Dataset
&nbsp;&nbsp;dataset_root:&nbsp;/home/aistudio/data/src/train
&nbsp;&nbsp;train_path:&nbsp;/home/aistudio/data/src/train.txt
&nbsp;&nbsp;num_classes:&nbsp;9
&nbsp;&nbsp;transforms:
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;ResizeStepScaling
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;min_scale_factor:&nbsp;0.5
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;max_scale_factor:&nbsp;2.0
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;scale_step_size:&nbsp;0
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;RandomHorizontalFlip
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;RandomVerticalFlip
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;Resize
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;target_size:&nbsp;[512,&nbsp;512]
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;RandomDistort
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;brightness_range:&nbsp;0.25
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;brightness_prob:&nbsp;1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;contrast_range:&nbsp;0.25
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;contrast_prob:&nbsp;1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;saturation_range:&nbsp;0.25
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;saturation_prob:&nbsp;1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hue_range:&nbsp;63
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hue_prob:&nbsp;1
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;Normalize
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mean:&nbsp;[0.485,&nbsp;0.456,&nbsp;0.406]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;std:&nbsp;[0.229,&nbsp;0.224,&nbsp;0.225]
&nbsp;&nbsp;mode:&nbsp;train


val_dataset:
&nbsp;&nbsp;type:&nbsp;Dataset
&nbsp;&nbsp;dataset_root:&nbsp;/home/aistudio/data/src/train
&nbsp;&nbsp;val_path:&nbsp;/home/aistudio/data/src/val.txt
&nbsp;&nbsp;num_classes:&nbsp;9
&nbsp;&nbsp;transforms:
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;Resize
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;target_size:&nbsp;[512,&nbsp;512]
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;Normalize
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mean:&nbsp;[0.485,&nbsp;0.456,&nbsp;0.406]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;std:&nbsp;[0.229,&nbsp;0.224,&nbsp;0.225]
&nbsp;&nbsp;mode:&nbsp;val


optimizer:
&nbsp;&nbsp;type:&nbsp;sgd
&nbsp;&nbsp;momentum:&nbsp;0.9
&nbsp;&nbsp;weight_decay:&nbsp;0.0001

lr_scheduler:
&nbsp;&nbsp;type:&nbsp;PolynomialDecay
&nbsp;&nbsp;learning_rate:&nbsp;0.005
&nbsp;&nbsp;end_lr:&nbsp;0
&nbsp;&nbsp;power:&nbsp;2

iters:&nbsp;160000

loss:
&nbsp;&nbsp;types:
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;CrossEntropyLoss
&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;type:&nbsp;CrossEntropyLoss
&nbsp;&nbsp;coef:&nbsp;[1,&nbsp;0.4]







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleSeg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年10月15日(星期五) 上午9:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleSeg] 请问mlatransformer应该如何使用呢 (#1441)





 
@huilin16 Hi, could you please show your config file?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by huilin16** [comment]
主要问题就是数据类型不对，跑mla transformer和segformer都是这个报错


将x.dtype改成keep_prob.dtype才可以运行











------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleSeg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年10月15日(星期五) 上午9:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleSeg] 请问mlatransformer应该如何使用呢 (#1441)





 
@huilin16 Hi, could you please show your config file?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by wenhwu** [comment]
@michaelowenliu , 请问mlatransformer是Paddleseg团队自己设计的吗？有没有相关论文呢，谢谢
